[identity] test failures after upgrading @azure/msal-node from 2.6.1 to 2.6.2

```diff
diff --git a/sdk/identity/identity/package.json b/sdk/identity/identity/package.json
index ae56d0809a8..ef40ba3530c 100644
--- a/sdk/identity/identity/package.json
+++ b/sdk/identity/identity/package.json
@@ -114,7 +114,7 @@
     "@azure/core-util": "^1.3.0",
     "@azure/logger": "^1.0.0",
     "@azure/msal-browser": "^3.5.0",
-    "@azure/msal-node": "^2.5.1",
+    "@azure/msal-node": "^2.6.2",
     "events": "^3.0.0",
     "jws": "^4.0.0",
     "open": "^8.0.0",
```

- Test failures
```
[node-tests]   1) IdentityClient
[node-tests]        throws an exception if the credential is not available (can't resolve discovery endpoint):
[node-tests]
[node-tests]       AssertionError: expected 'AuthenticationRequiredError' to equal 'CredentialUnavailableError'
[node-tests]       + expected - actual
[node-tests]
[node-tests]       -AuthenticationRequiredError
[node-tests]       +CredentialUnavailableError
[node-tests]
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/internal/identityClient.spec.ts:43:14)
[node-tests]       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[node-tests]
[node-tests]   2) IdentityClient
[node-tests]        throws an exception when an authentication request fails:
[node-tests]      AssertionError: expected false to be truthy
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/internal/identityClient.spec.ts:66:14)
[node-tests]       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[node-tests]
[node-tests]   3) OnBehalfOfCredential
[node-tests]        authenticates with a secret:
[node-tests]      AssertionError: expected undefined to be a number
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/internal/node/onBehalfOfCredential.spec.ts:49:12)
[node-tests]       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[node-tests]
[node-tests]   4) OnBehalfOfCredential
[node-tests]        authenticates with a certificate path:
[node-tests]      AssertionError: expected undefined to be a number
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/internal/node/onBehalfOfCredential.spec.ts:82:12)
[node-tests]       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[node-tests]
[node-tests]   5) ClientCertificateCredential
[node-tests]        allows cancelling the authentication:
[node-tests]
[node-tests]       AssertionError: expected 'AuthenticationRequiredError' to equal 'CredentialUnavailableError'
[node-tests]       + expected - actual
[node-tests]
[node-tests]       -AuthenticationRequiredError
[node-tests]       +CredentialUnavailableError
[node-tests]
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/public/node/clientCertificateCredential.spec.ts:100:12)
[node-tests]
[node-tests]   6) ClientSecretCredential
[node-tests]        allows cancelling the authentication:
[node-tests]
[node-tests]       AssertionError: expected 'AuthenticationRequiredError' to equal 'CredentialUnavailableError'
[node-tests]       + expected - actual
[node-tests]
[node-tests]       -AuthenticationRequiredError
[node-tests]       +CredentialUnavailableError
[node-tests]
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/public/node/clientSecretCredential.spec.ts:75:12)
[node-tests]       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[node-tests]
[node-tests]   7) UsernamePasswordCredential
[node-tests]        allows cancelling the authentication:
[node-tests]
[node-tests]       AssertionError: expected 'AuthenticationRequiredError' to equal 'CredentialUnavailableError'
[node-tests]       + expected - actual
[node-tests]
[node-tests]       -AuthenticationRequiredError
[node-tests]       +CredentialUnavailableError
[node-tests]
[node-tests]       at Context.<anonymous> (/home/meng/git/jsmain/sdk/identity/identity/test/public/node/usernamePasswordCredential.spec.ts:73:12)
[node-tests]       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[node-tests]
```

--- a/sdk/identity/identity/test/internal/identityClient.spec.ts
+++ b/sdk/identity/identity/test/internal/identityClient.spec.ts
@@ -31,7 +31,9 @@ describe("IdentityClient", function () {
   it("throws an exception if the credential is not available (can't resolve discovery endpoint)", async () => {
     const { error } = await testContext.sendCredentialRequests({
       scopes: ["scope"],
-      credential: new ClientSecretCredential(PlaybackTenantId, "client", "secret"),
+      credential: new ClientSecretCredential(PlaybackTenantId, "client", "secret", {
+        authorityHost: "https://unreachable-host.com",
+      }),
       secureResponses: [
         createResponse(400, {
           error: "test_error",

--- a/sdk/identity/identity/test/internal/identityClient.spec.ts
+++ b/sdk/identity/identity/test/internal/identityClient.spec.ts
@@ -181,7 +181,7 @@ describe("IdentityClient", function () {
       (msg: string) => msg.indexOf("azure:identity:") >= 0,
     );
 
-    assert.equal(expectedMessages.length, logMessages.length);
+    assert.equal(logMessages.length, expectedMessages.length);
 
     for (let i = 0; i < logMessages.length; i++) {
       assert.ok(logMessages[i].match(expectedMessages[i]), `Checking[${i}] ${logMessages[i]}`);

--- a/sdk/identity/identity/test/internal/identityClient.spec.ts
+++ b/sdk/identity/identity/test/internal/identityClient.spec.ts
@@ -32,6 +32,7 @@ describe("IdentityClient", function () {
     const { error } = await testContext.sendCredentialRequests({
       scopes: ["scope"],
       credential: new ClientSecretCredential(PlaybackTenantId, "client", "secret", {
+        // createResponse below will simulate a 400 error when trying to resolve the authority
         authorityHost: "https://unreachable-host.com",
       }),
       secureResponses: [
@@ -54,7 +55,11 @@ describe("IdentityClient", function () {
   it("throws an exception when an authentication request fails", async () => {
     const { error } = await testContext.sendCredentialRequests({
       scopes: ["https://test/.default"],
-      credential: new ClientSecretCredential("adfs", "client", "secret"),
+      credential: new ClientSecretCredential("adfs", "client", "secret", {
+        // createResponse below will simulate a 200 when trying to resolve the authority,
+        // then the 400 error when trying to get the token
+        authorityHost: "https://valid-authority.com",
+      }),
       secureResponses: [
         ...prepareMSALResponses(),
         createResponse(400, {

--- a/sdk/identity/identity/test/internal/identityClient.spec.ts
+++ b/sdk/identity/identity/test/internal/identityClient.spec.ts
@@ -33,7 +33,7 @@ describe("IdentityClient", function () {
       scopes: ["scope"],
       credential: new ClientSecretCredential(PlaybackTenantId, "client", "secret", {
         // createResponse below will simulate a 400 error when trying to resolve the authority
-        authorityHost: "https://unreachable-host.com",
+        authorityHost: "https://fake-authority.com",
       }),
       secureResponses: [
         createResponse(400, {
@@ -58,7 +58,7 @@ describe("IdentityClient", function () {
       credential: new ClientSecretCredential("adfs", "client", "secret", {
         // createResponse below will simulate a 200 when trying to resolve the authority,
         // then the 400 error when trying to get the token
-        authorityHost: "https://valid-authority.com",
+        authorityHost: "https://fake-authority.com",
       }),
       secureResponses: [
         ...prepareMSALResponses(),

--- a/sdk/identity/identity/test/internal/node/onBehalfOfCredential.spec.ts
+++ b/sdk/identity/identity/test/internal/node/onBehalfOfCredential.spec.ts
@@ -27,6 +27,7 @@ describe("OnBehalfOfCredential", function () {
       clientId: "client",
       clientSecret: "secret",
       userAssertionToken: "user-assertion",
+      authorityHost: "https://fake-authority.com",
     });
 
     const newMSALClientLogs = (): number =>
@@ -60,6 +61,7 @@ describe("OnBehalfOfCredential", function () {
       clientId: "client",
       certificatePath,
       userAssertionToken: "user-assertion",
+      authorityHost: "https://fake-authority.com",
     });
 
     const newMSALClientLogs = (): number =>

--- a/sdk/identity/identity/test/public/node/clientCertificateCredential.spec.ts
+++ b/sdk/identity/identity/test/public/node/clientCertificateCredential.spec.ts
@@ -74,6 +74,7 @@ describe("ClientCertificateCredential", function () {
       env.IDENTITY_SP_CLIENT_ID || env.AZURE_CLIENT_ID!,
       certificatePath,
       recorder.configureClientOptions({
+        authorityHost: "https://fake-authority.com",
         httpClient: {
           async sendRequest(): Promise<PipelineResponse> {
             await delay(100);

--- a/sdk/identity/identity/test/public/node/clientSecretCredential.spec.ts
+++ b/sdk/identity/identity/test/public/node/clientSecretCredential.spec.ts
@@ -5,6 +5,7 @@
 
 import { MsalTestCleanup, msalNodeTestSetup } from "../../node/msalNodeTestSetup";
 import { Recorder, delay, env, isRecordMode } from "@azure-tools/test-recorder";
+
 import { AbortController } from "@azure/abort-controller";
 import { ClientSecretCredential } from "../../../src";
 import { Context } from "mocha";
@@ -55,7 +56,9 @@ describe("ClientSecretCredential", function () {
       env.AZURE_TENANT_ID!,
       env.AZURE_CLIENT_ID!,
       env.AZURE_CLIENT_SECRET!,
-      recorder.configureClientOptions({}),
+      recorder.configureClientOptions({
+        authorityHost: "https://fake-authority.com",
+      }),
     );
 
     const controller = new AbortController();

--- a/sdk/identity/identity/test/public/node/usernamePasswordCredential.spec.ts
+++ b/sdk/identity/identity/test/public/node/usernamePasswordCredential.spec.ts
@@ -5,6 +5,7 @@
 
 import { MsalTestCleanup, msalNodeTestSetup } from "../../node/msalNodeTestSetup";
 import { Recorder, delay, env, isLiveMode } from "@azure-tools/test-recorder";
+
 import { AbortController } from "@azure/abort-controller";
 import { Context } from "mocha";
 import { UsernamePasswordCredential } from "../../../src";
@@ -54,6 +55,9 @@ describe("UsernamePasswordCredential", function () {
       clientId,
       env.AZURE_IDENTITY_TEST_USERNAME || env.AZURE_USERNAME!,
       env.AZURE_IDENTITY_TEST_PASSWORD || env.AZURE_PASSWORD!,
+      recorder.configureClientOptions({
+        authorityHost: "https://fake-authority.com",
+      }),
     );
 
     const controller = new AbortController();

